### PR TITLE
Add Hero C Compiler to the vendor list & add C source language

### PR DESCRIPTION
**Pull request by heroseh**

### CODE REVIEW

**Comment by alan-baker** [comment]
It looks like you haven't regenerated the headers (e.g. spirv.h). See the instructions [here](https://github.com/KhronosGroup/SPIRV-Headers#generating-headers-from-the-json-grammar-for-the-spir-v-core-instruction-set).

File: include/spirv/unified1/spirv.core.grammar.json [hunk 1]
```diff
@@ -10065,6 +10065,10 @@
         {
           "enumerant" : "SYCL",
           "value" : 7
+        },
+        {
+          "enumerant" : "C",
```
Comment (Hugobros3) [inline include/spirv/unified1/spirv.core.grammar.json:10070]: This addition was discussed during the last SPIR-V call today, here's a summary:

We have already various language entries for OpenCL C and C++ flavours, so perhaps "C" is a bit too generic. Both in the sense that it appears you are implementing a subset of C, but also you have your own extensions for exposing things like textures etc which do not exist in standard C. Perhaps it makes sense to define your own language for your dialect of C, and create a language entry for that ?

**Comment by bashbaug** [comment]
Ping @heroseh - just want to be sure you saw the comments above.  Let us know if there is anything we can do to help.  Thanks!

**Comment by heroseh** [comment]
Sorry for the delayed response, thank you all for taking the time to review this! I have made the requested adjustments. Is it all good now?

**Pull request merged**